app/models/transfer.rb

```diff
@@ -99,18 +105,13 @@ def transfer_account
     end
 
     def bank_account
-     @bank_account ||= BankAccount.find(@bank_account_id)
+     @bank_account ||= BankAccount.find_by_id(@bank_account_id)
```

Comment:
@Gataniel use `bind_by(id: ...)`. And use accessor methods instead class variables: `bank_account_id` instead `@bank_accont_id`